x-pack/plugins/ml/public/application/components/job_selector/job_selector_flyout.tsx

```diff
@@ -71,11 +75,15 @@ export const JobSelectorFlyoutContent: FC<JobSelectorFlyoutProps> = ({
     },
   } = useMlKibana();
 
+  const [applyTimeRangeState, setApplyTimeRangeState] = useState<boolean>(true);
+
+  const handleToggleApplyTimeRange = setApplyTimeRangeConfig ?? setApplyTimeRangeState;
+  const applyTimeRange = applyTimeRangeConfig ?? applyTimeRangeState;
```

Comment:
I believe mixing higher-order state and local state is error-prone and more difficult to maintain. Would it be better if the parent component always controls it?

Reply:
Updated here [`82b3c33` (#97407)](https://github.com/elastic/kibana/pull/97407/commits/82b3c330d69d78c0386ca7164db80736b36ffe0b)